Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,37 +1,107 @@
+#include <math.h>
+#include <QTime>
+
 #include "BatteryStateOfChargeService.h"
+#include "BatteryData.h"
 
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+    const int MS_IN_HOURS = 3600000;
+    const int MS_IN_MINUTES = 60000;
+    const int MS_IN_SECONDS = 1000;
 }
 
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+    /*
+                : initialStateOfChargePercent_(initialStateOfChargePercent)
+      Same as ^   initialStateOfChargePercent_ = initialStateOfChargePercent;
+      Another example in BatteryData.cpp                                     */
 }
 
 BatteryStateOfChargeService::~BatteryStateOfChargeService()
 {
 }
 
+/*Returns the total amp hours used*/
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpHoursUsed_;
 }
 
+/* The battery is charging if the battery current is negative
+ * The battery is depleting if the battery current is positive
+ * Returns true if the battery is being charged, false if not.*/
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (currentNow_ < 0)
+    {
+        return true;
+    } else
+    {
+        return false;
+    }
```

Comment:
To make this code even shorter, write return(currentNow_ < 0). This returns a boolean